### ISSUE TITLE
Component downloads previous state of props with redux

### ISSUE THREAD

**Issue by ricardoromero4444** [react-csv/react-csv#209]
I am using the component as follows:
```javascript
<CSVLink data={users}>
  <Button
    color="primary"
    variant={'contained'}
    disabled={usersLoading}
  >
    Sheet
  </Button>
</CSVLink>
```
Whenever I click the component, it downloads a CSV file of the previous `users` value that was passed to it (not the current `users`!). 
`Users` variable comes from redux store, and is being updated properly. (I checked printing with `onClick()`. It seems the component has access to the current `users` value but is still downloading the previous one)
After the first render, it will download an empty file when clicked, as `users` was previously `undefined`. After the second render, it will download a file with what would have been the correct first value for `users` and so on.
UPDATE: Reverting to 1.1.2 solves the issue.

**Comment by mishabhi** [comment]
I am also facing the same issue with the current version.

**Comment by champunes** [comment]
I've the same problem. It seems related with this commit: 75c2ba4d11e6681ee1a5bd1962eea68a46e38532

**Comment by ChristianKis** [comment]
I faced the same issue, props are not being updated properly in the component. 

In a simple example, binding some state data from the container component to CSVLink's data property, it will download the previous state of the bound data.

I can also verify that reverting to 1.1.2 fixed the issue.

**Comment by janeeisenstein** [comment]
Same here. So glad a documented work around exists.

**Comment by danmun** [comment]
I just came across this issue too. 
I was using CSVLink as such:
```
<Button as={CSVLink} 
        data={analysis.data}
        filename={moment(Date.now()).format('YYYYMMDDHHMM')+ "_" + analysis.id + "_proteins.csv"}>
                <Icon name='file excel'/>
                Download CSV
</Button>
```
The state was updating properly but the CSVLink downloaded an older version of state.
Downgrading from 2.0.1 to 1.1.2 solved the issue, thanks all.

**Comment by abdennour** [comment]
Thank you guys,
🎉🎉🎉🎉🎉🎉🎉
It is now merged and the new version is [2.0.2](https://www.npmjs.com/package/react-csv/v/2.0.2)

https://github.com/react-csv/react-csv/releases/tag/v2.0.2
